{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 110,
   "metadata": {},
   "outputs": [],
   "source": [
    "from pyensembl import EnsemblRelease as ensbl\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "import argparse\n",
    "import os"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Utility functions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 119,
   "metadata": {},
   "outputs": [],
   "source": [
    "def parseArgs(description = 'Get coverage at different DPs for a given list of genes.'):\n",
    "    parser = argparse.ArgumentParser(description= description)\n",
    "    parser.add_argument('-b','--bamfile',required=True,help='input bam file')\n",
    "    parser.add_argument('-g','--genelistfile',required=True)\n",
    "    parser.add_argument('-o','--outpath',required=True)\n",
    "    parser.add_argument('-r','--ref',dest = 'ref',default = '',required = True,type=int,help = 'Reference genome is mandatory since only a list of genames is given and we need to match it inside our bam file')\n",
    "    parser.add_argument('-v','--vcffile',default = '',required=False,help='input vcf file to be usde for complementary analysis', )\n",
    "    parser.add_argument('-p','--prefix',dest = 'prefix',default = '',required = False)\n",
    "    parser.add_argument('-s','--splitBamFile', dest='split', action='store_false',help = 'in next iterations this flag will allow to split the reducedBAM file by chromosomes')\n",
    "    parser.set_defaults(split=False)\n",
    "\n",
    "    args = parser.parse_args()\n",
    "    # MANDATORY ARGUMENTS\n",
    "    bamfile = args.bamfile\n",
    "    vcffile = args.vcffile\n",
    "    outpath = args.outpath\n",
    "    prefix = args.prefix\n",
    "    ref = args.ref\n",
    "    genelistfile = args.genelistfile\n",
    "    \n",
    "    #OPTIONAL ARGUMENTS\n",
    "    split= args.split\n",
    "    \n",
    "    return bamfile, vcffile, outpath, prefix, ref, genelistfile, split\n",
    "    \n",
    "    \n",
    "    \n",
    "def check_ref(ref,outpath):\n",
    "    if ref not in [19,37,38]:\n",
    "        print 'please, reference must be one of 19,37,38\\n'\n",
    "        print 'bye'\n",
    "        quit()\n",
    "    else:\n",
    "        if ref == 19:\n",
    "            print 'using ref hg19, '\n",
    "        if ref == 37:\n",
    "            print 'using ref hgrc37'\n",
    "        if ref == 38:\n",
    "            print 'using ref hgrc38'\n",
    "    print ''\n",
    "    \n",
    "    if not os.path.isdir(outpath):\n",
    "        os.system('mkdir %s'%outpath)    \n",
    "\n",
    "    \n",
    "    \n",
    "def get_genelist(genelistfile):\n",
    "    genes = pd.read_csv(genelistfile,header = None)\n",
    "    genelist = genes.iloc[:,0].tolist()\n",
    "    print 'genes read from file:'\n",
    "    print genelist\n",
    "    print '\\n'\n",
    "\n",
    "    return genelist\n",
    "    \n",
    "    \n",
    "def test_mode():\n",
    "    ref = 37\n",
    "    genelistfile = './test/genelist'\n",
    "    bamfile = './test/test_chr2_190-192k.bam'\n",
    "    outpath = './test_output/'\n",
    "    split = False\n",
    "    return(ref,genelistfile,bamfile,outpath,split)\n",
    "    \n",
    "    \n",
    "def get_locis(genelist,ref,outpath,write_bedfile = True):\n",
    "    analyzed_genes = []\n",
    "    try:\n",
    "        if (ref == 37)|(ref == 19):\n",
    "            ensemble_version = 75\n",
    "    except:\n",
    "        'You must to install the ensemble version 75, please run \\n'        \n",
    "        'pyensembl install --release 75 --species homo_sapiens \\n'\n",
    "    try:\n",
    "        if ref == 38:\n",
    "            ensemble_version = 87\n",
    "    except:\n",
    "        'You must to install latest ensemble version (87), please run \\n'        \n",
    "        'pyensembl install --release 87 --species homo_sapiens \\n'\n",
    "        \n",
    "    data = ensbl(ensemble_version)\n",
    "    locis = []\n",
    "\n",
    "    for gen in genelist:\n",
    "        try:\n",
    "            loc =  data.loci_of_gene_names(gen)\n",
    "            locis.append(loc[0].to_dict())\n",
    "            analyzed_genes.append(gen)\n",
    "        except:\n",
    "            print 'warning, gen %s was not found and ignored'%gen\n",
    "#    return pd.concat(locis)\n",
    "    print '\\n'\n",
    "    ggenes = analyzed_genes\n",
    "    locis = pd.DataFrame(locis)\n",
    "    locis['name'] = ggenes\n",
    "    locis.rename(columns = {'contig':'chrom'},inplace = True)\n",
    "    locis['score'] = 0 ## this is only for completness, in order to recognize strand as the right next field\n",
    "    locis = locis[['chrom','start','end','name','score','strand']]\n",
    "    \n",
    "    # write file\n",
    "    bedfile = outpath+'gene_loci.bed'\n",
    "    if write_bedfile:\n",
    "        locis.to_csv(bedfile,sep = '\\t',index = False,header = None)\n",
    "        \n",
    "    return locis, bedfile, locis.columns\n",
    "\n",
    "\n",
    "def run_samtools_view(bamfile,bedfile,split=False):\n",
    "    base = os.path.basename(bamfile).split('.bam')[0]\n",
    "    #print base\n",
    "    outbam = outpath + base + '_reduced.bam'\n",
    "    if not split:\n",
    "        bedfile = bedfile\n",
    "        call1 = 'samtools view -bh -L %s %s > %s'%(bedfile,bamfile,outbam)\n",
    "        #print call1\n",
    "    status = os.system(call1)    \n",
    "    if status ==0:\n",
    "        print 'ok samtools'\n",
    "    else:\n",
    "        print 'samtools have failed'\n",
    "        #quit()\n",
    "\n",
    "    # get index\n",
    "    call2 = 'samtools index %s'%(outbam)\n",
    "    index = os.system(call2)\n",
    "    if index ==0:\n",
    "        print 'ok index'\n",
    "    else:\n",
    "        print 'index have failed'\n",
    "        #quit()\n",
    "\n",
    "    return  outbam\n",
    "    \n",
    "#def clean_files(): \n",
    "#    rm \n",
    "\n",
    "def serie_counting(x):\n",
    "    return pd.Series({'bp_at_10dp':(x>10).sum(),'bp_at_20dp':(x>20).sum(),'bp_at_30dp':(x>30).sum()})\n",
    "\n",
    "def run_bedtools_coverage(bam,bed,outpath):\n",
    "    outcoveragefile = outpath + 'coverage_by_gen.tsv'\n",
    "    call = 'coverageBed -a %s -b %s > %s'%(bed,bam,outcoveragefile)\n",
    "    status = os.system(call)\n",
    "    #print call\n",
    "    if status ==0:\n",
    "        print 'ok coveragedBed'\n",
    "    else:\n",
    "        print 'coveragedBed have failed'\n",
    "\n",
    "    results = pd.read_table(outcoveragefile,header = None)\n",
    "    results.columns = ['chr','start','end','name','score','strand','matched_reads','coverage_dp1','len_gen_bp','rel_coverage_dp1']\n",
    "    \n",
    "    \n",
    "    ### the later approach give us the coverage as the \n",
    "    #number of read bases at least ONE time over the total length of our gene of interest\n",
    "    \n",
    "    ## in the following I atempt to compute the coverage at different minimal dp levels, namely: at dp= 10, 20, 30\n",
    "    \n",
    "    #compute dp per base x gene\n",
    "    file_by_position = outpath + 'coverage_by_position.txt'\n",
    "    call2 = 'coverageBed -a %s -b %s -d > %s'%(bed,bam,file_by_position)\n",
    "    os.system(call2)\n",
    "    \n",
    "    bypos = pd.read_csv(file_by_position,sep = '\\t',header = None)\n",
    "    #print bypos.head()\n",
    "    grouping = bypos.groupby([3])[7].apply(lambda x: serie_counting(x))\n",
    "    \n",
    "    res  = grouping.unstack(); \n",
    "    #reset index and rename it\n",
    "    cols = res.columns  #reserve columns\n",
    "    res.reset_index(inplace = True)\n",
    "    res.columns = ['name']+list(cols)\n",
    "\n",
    "    \n",
    "    ###  merge both analysis\n",
    "    coverage = pd.merge(results,res, on = 'name')\n",
    "    \n",
    "    ### relativise results\n",
    "    abscols = ['bp_at_10dp','bp_at_20dp','bp_at_30dp']\n",
    "\n",
    "    relatives = np.round((coverage[abscols].transpose() /coverage['len_gen_bp'].values).transpose(),7)\n",
    "    coverage[abscols] = relatives\n",
    "    coverage.to_csv(outcoveragefile,index = False, sep = '\\t')\n",
    "\n",
    "    \n",
    "    \n",
    "    return outcoveragefile"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 118,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "using ref hgrc37\n",
      "\n",
      "genes read from file:\n",
      "['STAT1', 'HLA-A', 'TP53', 'CANDF7']\n",
      "\n",
      "\n",
      "warning, gen CANDF7 would be ignored\n",
      "\n",
      "\n",
      "samtools view -bh -L ./test_output/gene_loci.bed ./test/test_chr2_190-192k.bam > ./test_output/test_chr2_190-192k_reduced.bam\n",
      "ok samtools\n",
      "ok index\n",
      "ok coveragedBed\n"
     ]
    }
   ],
   "source": [
    "#bamfile, vcffile, outpath, prefix, ref, genelistfile, split = parseArgs()    \n",
    "if(False):\n",
    "    ref, genelistfile, bamfile, outpath, split = test_mode()\n",
    "\n",
    "    #check params\n",
    "    check_ref(ref=ref,outpath = outpath)\n",
    "\n",
    "    genelist = get_genelist(genelistfile)\n",
    "    gene_loci , bedfile , locicolumns = get_locis(genelist,ref = ref,outpath=outpath)\n",
    "        \n",
    "    reduced_bamfile = run_samtools_view(bamfile,bedfile,split=split)\n",
    "\n",
    "    #compute coverage by gen along the bamfile\n",
    "    coverage_file = run_bedtools_coverage(reduced_bamfile,bedfile,outpath)\n",
    "    \n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "if __name__ == \"__main__\":\n",
    "    #parse arguments\n",
    "    bamfile, vcffile, outpath, prefix, ref, genelistfile, split = parseArgs()    \n",
    "\n",
    "    #check params\n",
    "    check_ref(ref=ref,outpath = outpath)\n",
    "\n",
    "    genelist = get_genelist(genelistfile)\n",
    "    gene_loci , bedfile , locicolumns = get_locis(genelist,ref = ref,outpath=outpath)\n",
    "        \n",
    "    reduced_bamfile = run_samtools_view(bamfile,bedfile,split=split)\n",
    "\n",
    "    #compute coverage by gen along the bamfile\n",
    "    coverage_file = run_bedtools_coverage(reduced_bamfile,bedfile,outpath)\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "<matplotlib.collections.PathCollection at 0x7f76dce18690>"
      ]
     },
     "execution_count": 42,
     "metadata": {},
     "output_type": "execute_result"
    },
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAXoAAAD8CAYAAAB5Pm/hAAAABHNCSVQICAgIfAhkiAAAAAlwSFlz\nAAALEgAACxIB0t1+/AAAGHlJREFUeJzt3X2MHdV9xvHvj/VLzEtiOziWMW7WOBQEggJZAamjijSl\nEIiK0yoRFkpQiuKoIWoQEe0aUAEJFDc0JEFtSUyDAgIMjoCFYIrjkFRpUjBZY2NjiIsBg70Ye/Pi\n4MIG2+tf/7hn13eXve937syceT7SamfOzL33zO7c5849c+aMuTsiIhKvw9KugIiIJEtBLyISOQW9\niEjkFPQiIpFT0IuIRE5BLyISOQW9iEjkFPQiIpFT0IuIRG5S2hUAOProo727uzvtaoiI5Mq6det+\n7e6zaq1XM+jNbB5wFzAbcGC5u3/bzK4HvgAMhlWvdvfHwmOWApcBw8Dfu/vqaq/R3d1Nf39/raqI\niEgZM3u1nvXqOaI/AHzV3Z8xs6OAdWa2Jiz7prv/y7gXPgm4GDgZOAb4sZn9sbsP1199ERFpl5pt\n9O6+092fCdN7gReAuVUechFwn7u/4+6vAFuBM9tRWRERaVxDJ2PNrBs4HVgbir5sZhvN7A4zmxHK\n5gLbyx62g+ofDCIikqC6g97MjgQeAK5w9zeB24AFwGnATuAbjbywmS0xs34z6x8cHKz9ABERaUpd\nQW9mkymF/D3u/iCAu+9y92F3PwjczqHmmQFgXtnDjw1lY7j7cnfvcfeeWbNqnjQWEZEm1dPrxoDv\nAS+4+y1l5XPcfWeY/RTwXJh+BLjXzG6hdDL2eODpttZa3qVv/QA3r97C63uGOGb6NK467wQWna4W\nMxGpr9fNQuCzwCYz2xDKrgYWm9lplLpcbgO+CODum81sJfA8pR47l6vHTbL61g9wxf0bRucH9gyN\nzivsRcSycCvBnp4eVz/65s3vXcVE/0UDXll2YaerIyIdYmbr3L2n1noaAiEClT6q0/8IF5EsUNCL\niEROQS8iEjkFvYhI5BT0IiKRU9CLiEROQS8iEjkFvYhI5BT0IiKRU9CLiEROQS8iEjkFvYhI5BT0\nIiKRU9CLiEROQS8iEjkFvYhI5BT0IiKRU9CLiEROQS8iEjkFvYhI5BT0IiKRU9CLiEROQS8iEjkF\nvYhI5BT0IiKRU9CLiEROQS8iEjkFvYhI5BT0IiKRU9CLiEROQS8iEjkFvYhI5GoGvZnNM7Ofmtnz\nZrbZzL4Symea2RozezH8nhHKzcxuNbOtZrbRzM5IeiNERKSyeo7oDwBfdfeTgLOBy83sJKAXeMLd\njweeCPMAnwCODz9LgNvaXmsREalbzaB3953u/kyY3gu8AMwFLgLuDKvdCSwK0xcBd3nJU8B0M5vT\n9pqLiEhdGmqjN7Nu4HRgLTDb3XeGRW8As8P0XGB72cN2hLLxz7XEzPrNrH9wcLDBaouISL3qDnoz\nOxJ4ALjC3d8sX+buDngjL+zuy929x917Zs2a1chDRUSkAXUFvZlNphTy97j7g6F410iTTPi9O5QP\nAPPKHn5sKBMRkRTU0+vGgO8BL7j7LWWLHgEuDdOXAg+XlX8u9L45G/h9WROPiIh02KQ61lkIfBbY\nZGYbQtnVwDJgpZldBrwKfCYsewy4ANgKvA18vq01FhGRhtQMenf/OWAVFn98gvUduLzFeomISJvo\nylgRkcgp6EVEIqegFxGJnIJeRCRyCnoRkcgp6EVEIqegFxGJnIJeRCRyCnoRkcgp6EVEIqegFxGJ\nnII+ArOPmtJQuYgUi4I+ApO6uhoqF5FiUdBHYGDPUEPlIlIsCnoRkcgp6EVEIqegFxGJnIJeRCRy\n9dwzVkSko/rWD3Dz6i28vmeIY6ZP46rzTmDR6XPTrlZuKegL6Nq+TaxYu51hd7rMWHzWPG5cdEra\n1RIBSvvn3U+9Njo/sGeIK1duAFDYN0lNNwUz8iYadgdg2J27n3qNa/s2pVwzkdKRfHnIjzjocPWD\nG1OoURwU9AUz0ZuoWrlIJ13/yOaKy97ef7CDNYmLgj4CMw6f3FC5SFbtGdqfdhWipKCPwP/9YeI3\nR6VyESkWBX0EKn2j1TddEQEFvYhI9BT0IiKRU9CLiEROF0yJSCrOumkNu/buG53XjXKSoyN6Eem4\n8SEPvGte2kdBLyIdp1DvrJpBb2Z3mNluM3uurOx6Mxswsw3h54KyZUvNbKuZbTGz85KquIiI1Kee\nI/rvA+dPUP5Ndz8t/DwGYGYnARcDJ4fH/LuZ6calIiIpqhn07v4z4Ld1Pt9FwH3u/o67vwJsBc5s\noX4iItKiVtrov2xmG0PTzoxQNhfYXrbOjlAmIiIpaTbobwMWAKcBO4FvNPoEZrbEzPrNrH9wcLDJ\nakgtGn5YRJoKenff5e7D7n4QuJ1DzTMDwLyyVY8NZRM9x3J373H3nlmzZjVTDanDPRp+WKTwmgp6\nM5tTNvspYKRHziPAxWY21czmA8cDT7dWRWmFp10BEUldzStjzWwFcA5wtJntAK4DzjGz0yjlyDbg\niwDuvtnMVgLPAweAy919OJmqi4hIPWoGvbsvnqD4e1XWvwm4qZVKSf0uuf3Jqstt3Px7p3bx5jvv\n/ux971T1ghWJla6MzblfvFS95+v4ppuJQr5auYjkn4I+cuOP6EWkeBT0kdPJWBFR0IuIRE5Bn3OT\na/wHD1PbjWRM3/oJL62RBCnoc+7mT59WdflBtd1IxlzzkK7W7jTdYSoCRuW2+LnTp3WyKqOu7dvE\nirXbGXany4zFZ83jxkWnpFIXyZa39qmHV6cp6HPumoc2VQz5aZO7uOq8EzpaHyiF/N1lQy8Mu4/O\nK+xFOk9NNzlX7ejoa399CotO7/zgoXdXGF+nUrkUi04bdZ6CXkQ66k8XzEy7CoWjoM+5akdHV/3g\nWfVwkMx5fufetKtQOAr6nKt2dLT/oHP9I5s7WBuR2n739v60q1A4Cvqcq3V0tGcoW28q3QhFpPMU\n9DmXt6OjFWu3116p4PrWD7Bw2U+Y37uKhct+ouY3aZm6V0Yua1fGDruu4Kqmb/0AV9y/YXR+YM/Q\n6HwaPagkDjqij1zWrozN2OdO5pSHfD3lRaJ7JjRPQR+5rP2DM/a5IzmieyY0L2s5IA2afdSUqssP\ndqgeIpJdaqPPmfFjyJx93Ax27a1+lykRKTYFfY5MNIZMrVsJisRiUtZ6FuSImm5yRGPFSAymdDUX\n2Aey1rMgR3REn2GX3P5kW4/YdbGSZMG+4eYCu8t0RN8sBX1GtTvkIZmLlfrWD3Dz6i28vmeIY6ZP\nS2VYZCkGXYPRPAV9RiXR9t7uN0q1i3tE2k1H9M1TG32BtPuNcqVCXTpIR/TNU9AXyNnHzai4rJmP\nAPXRF8kHNd0UyLbfDFVcNj7oJ2p711grIvmkoC+QgT2Vg7786Lxv/QBLH9zE0P7h0cctfbDUY0dh\nL5I/arqJ3OQm/sM3r94yGvIjhvYPc/PqLW2qlYh0koI+cvubaEivdORf7RuBiGSXgl5EJHIKepGM\nKMqdpBQ6nVfzb25md5jZbjN7rqxsppmtMbMXw+8ZodzM7FYz22pmG83sjCQrL/mk2+NN7Ksri3Fd\ngrrldl49H67fB84fV9YLPOHuxwNPhHmATwDHh58lwG3tqabEZOQKWoX9WE0OASNSU82gd/efAeOv\nx78IuDNM3wksKiu/y0ueAqab2Zx2VVYat3DBzLSrUJGGSyimZkevlOY121w22913huk3gNlhei5Q\nPnLWjlAmKfl0zx+lXQWRMZodvVKa1/J5EXd3mrgVqJktMbN+M+sfHBxstRpSwVU/0FGzSNE1G/S7\nRppkwu/doXwAmFe23rGh7F3cfbm797h7z6xZs5qshtTSTD96SYduoFSbzus0p9mgfwS4NExfCjxc\nVv650PvmbOD3ZU08krJaOXLJ7U92pB7lru3bxIKlj9Hdu4oFSx8r9M1Rqt1Aafq0yZ2rSIbpvE5z\nao51Y2YrgHOAo81sB3AdsAxYaWaXAa8CnwmrPwZcAGwF3gY+n0CdpUm12tfSuP/s+HvgjszfuOiU\njtcly/YdGK69kkgFNYPe3RdXWPTxCdZ14PJWKyXtM3XSoS9tXWa5GNP77qdeU9CP87ba4KQFGr0y\ncvsOHAqIPIT8iL71AxopcwLze1eN+WZmwCvLLkyrOpITCvrIHTN92uh0Xo7ooXT3KgX9WONDHkrN\ncfN7V42GfXfvqnc9bps+CApPw05E7mMnHurRlJeQh2JeJl/ezDaRSv+9kfKJQr5auRSHgj5yjz57\nqNPT3LKje8medw4U8eNNOkFBH7k9Q/tHp6867wSmTe5KsTZSiY66JUkK+gJZdPpc/ubDrbV7d7qf\ne9/6ARYu+wnze1dFO+pljNsk2aKgL5C+9QPc/8vttVesorzfe9L61g9w5coNDOwZwimNennlyvhG\nvbzmoeJeJCadoaAvkBt+uJn9VQaUqnUysNP+8YGN77pa9KCXymPy1j5dDCXJytY7WxL1u7f3V13e\n88HpHapJfSqdnNRJS5HGqB+9jEpjCITY9K0f4ObVW3h9zxDHTJ/GVeedoOsBJHUKemlKbO3k7dC3\nfmDMoFsjd9ICFPaSKjXdRC6pvvM3/HBzIs+bZ5VGVszCiItFHhVUFPTR635/MkFfq71fsuWeDvaW\nkuxR0EfuqZd/V/e6R0zRxVSxys/gF5IEBX3k6h3fpusw46ZPaWhgkRgp6AWAxWfO0wlDkUgp6AWA\ne596TT1pRCKloC+QajefPggsfbD+K051H2uR/FDQZ9Thk9v/r6l282mAof0H6x5FUSf3RPJDQS/S\nAWfdtCbtKkRhSpe+SzZDV8ZmVBI3g54+bfKY8emlc3bt3Zd2FTouieEg9lUZlE8qU9AXyMnHHKXx\nbKQjqg0HIZ2nppsCUchLp2R5OIgiUtCLiEROTTdSWKde9zhvvnPoph/vndrFxhvOT7FGIsnQEX3k\n1EdhYuNDHuDNd4Y59brHE3k9/R8kTQr6yL1v2uS0q5BJ40O+Vnmr1FdE0qSgj5zpUFKk8BT0kdO4\n8RKT6fqG2hSdjM2oI6Z08da+ZJoRiuDavk2sWLudYXe6zFh81jxuXKRhmPNu3wG9J5qhoM8ohXzz\nru3bxN1ld1Qadh+dV9jnWxJXjBeBmm4y6MRrHku7Crl279qJb5tXqVwkdi0d0ZvZNmAvMAwccPce\nM5sJ3A90A9uAz7h7/fezE/6g8TxaUmmUzlqjd4rEqh1H9B9z99PcvSfM9wJPuPvxwBNhXkQSVOum\nMV3qflVoSTTdXATcGabvBBYl8BoiUubKGmPI1HvvYIlTq0HvwI/MbJ2ZLQlls919Z5h+A5g90QPN\nbImZ9ZtZ/+DgYIvVkCLRiaV3q3WKMoH72EiOtNrr5qPuPmBmHwDWmNmvyhe6u5vZhIcS7r4cWA7Q\n09Ojww2pm/pdNE6dVYqtpc95dx8Iv3cDDwFnArvMbA5A+L271UpK9ixcMDPtKiTmktufTLsKIm3V\ndNCb2RFmdtTINPCXwHPAI8ClYbVLgYdbraRkT8xj28e8bVJMrRzRzwZ+bmbPAk8Dq9z9cWAZcK6Z\nvQj8RZiXBqh/RHJq9U4RiVHTbfTu/jLwJxOU/wb4eCuVKjqdsEjOFfdvaPm+pe12GDrvIMnSufgM\n0hF9saiLe/0OV/ehpuivlkE6oi+WdlwIXZQLoqZO7kq7CrmkoJemaMfJlqJcEKVht5uj0Stb9KGl\nqzhQ9h6bZLD1axemV6EOUZtytsydPo2BPUNpVwNQ99QsUtC3YHzIAxzwUnkRwr6IsjrO/esZCXlQ\n99Qs0jfwFowP+VrlaZumE1ktGRnnfqSZZGSc+2v7NqVcM53Xker0zk9IFt784w0V4Dr4s25ak9hz\nl9/MpJ5ykaxQ0CdEb/507Nq7L+0qNCSLBwQSHwV95LqK0esut1as3Z52FaQAFPSR082q2iOpoROK\n0i1S0qVeN1I49YZ2d++qhGsi0hk6opfCueGHm2uu02jIa7A0yTIFveRSKxflJHF15TUP6aSqZJeC\nXnIpaxflvLVvOO0qiFSkoBcRiZyCXkQkcgp6EZHIKehFRCKnoBcRiZwumBIpuPm9q8aMfmnAK8s0\nzHZMdEQvUmDjQx5KQx7rquC4KOhFCqzaSDunXvd4w8+nD4hsUtONiEzozXcauwisaCE/0fZuy2iT\nl47opWG6J6iM96GlCvlq5WlT0Ceou3fV6E9Msjb8gNQnyZucZPX2mVKippsmNRre5etn9eudJGtk\nH1i4YCb3fOEjHX/9FWu3Z+JG5nmXx7uCKehT0N27SmHfJh9auip3R5O/eOm3XHL7kx0P+3bf5CSt\nb6ppt43n8TaharqRpkw6LP17FOYx5Efkvfkra82RWatP1kR3RD/RP3ySwdav6Qi6nQ4cTD9h8xry\neTLyfio/Yj73lv9KqTadkfY3hiREFfSVPtUPeOnob3zYx/gPnUgazQRSWxpHoc2+Znlz44u732pn\nldqmHU2i1XrT5Dkbogr6asqP/qrt7GmHfxJv/l+89Ft9tZVCqLafb1t24YTL6x3yod6wH/8aWfiA\nSCzozex84NtAF/Af7r4sqddqRDOBl8SnuYJX8ubEax7jD8P5bS+r9J4bGfIhqUAe/9xn3bSGXXv3\nTbhuUnUwb/OZeAAz6wL+FzgX2AH8Eljs7s9PtH5PT4/39/c39BoKShGJUSNhb2br3L2n1npJ9bo5\nE9jq7i+7+z7gPuCidj25Ql5EpH5JBf1cYHvZ/I5QJiIiHZZaP3ozW2Jm/WbWPzg4mFY1RESil1TQ\nDwDzyuaPDWWj3H25u/e4e8+sWbMSqoZIe2WhB4VIo5IK+l8Cx5vZfDObAlwMPJLQa4mISBWJBL27\nHwC+DKwGXgBWuvvmdj2/jqqk3bqs9pAOI/td0vvftmUXRrePN7o9s4+aEt3foB656l7ZqGa6V4qI\nFF3a3StFRCQjFPQiIpFT0IuIRE5BLyISOQW9iEjkMtHrxswGgVebfPjRwK/bWJ2s0Hbli7YrP2La\npg+6e80rTjMR9K0ws/56uhfljbYrX7Rd+RHjNtWiphsRkcgp6EVEIhdD0C9PuwIJ0Xbli7YrP2Lc\npqpy30YvIiLVxXBELyIiVeQ66M3sfDPbYmZbzaw37fpMxMzuMLPdZvZcWdlMM1tjZi+G3zNCuZnZ\nrWF7NprZGWWPuTSs/6KZXVpW/mEz2xQec6tZHcMwtr5N88zsp2b2vJltNrOvRLJd7zGzp83s2bBd\nN4Ty+Wa2NtTl/jD0NmY2NcxvDcu7y55raSjfYmbnlZWnss+aWZeZrTezRyPapm1hH9lgZv2hLNf7\nYGLcPZc/QBfwEnAcMAV4Fjgp7XpNUM8/A84Anisr+zrQG6Z7gX8O0xcA/wkYcDawNpTPBF4Ov2eE\n6Rlh2dNhXQuP/UQHtmkOcEaYPorSjeBPimC7DDgyTE8G1oY6rAQuDuXfAf4uTH8J+E6Yvhi4P0yf\nFPbHqcD8sJ92pbnPAlcC9wKPhvkYtmkbcPS4slzvg4n9rdKuQAv/5I8Aq8vmlwJL065Xhbp2Mzbo\ntwBzwvQcYEuY/i6wePx6wGLgu2Xl3w1lc4BflZWPWa+D2/cwcG5M2wUcDjwDnEXp4ppJ4/c7Svdb\n+EiYnhTWs/H74sh6ae2zlO7w9gTw58CjoY653qbwWtt4d9BHsw+28yfPTTd5vgH5bHffGabfAGaH\n6UrbVK18xwTlHRO+2p9O6eg399sVmjg2ALuBNZSOVvd46WY64+syWv+w/PfA+2l8e5P2LeAfgINh\n/v3kf5sAHPiRma0zsyWhLPf7YBImpV2BonN3N7Ncdn0ysyOBB4Ar3P3N8ibMvG6Xuw8Dp5nZdOAh\n4MSUq9QSM/sksNvd15nZOWnXp80+6u4DZvYBYI2Z/ap8YV73wSTk+Yi+5g3IM2yXmc0BCL93h/JK\n21St/NgJyhNnZpMphfw97v5gKM79do1w9z3ATyk1TUw3s5GDovK6jNY/LH8f8Bsa394kLQT+ysy2\nAfdRar75NvneJgDcfSD83k3pQ/lMItoH2yrttqMW2ucmUTpxMp9DJ4FOTrteFerazdg2+psZe8Lo\n62H6QsaeMHo6lM8EXqF0smhGmJ4Zlo0/YXRBB7bHgLuAb40rz/t2zQKmh+lpwH8DnwR+wNgTl18K\n05cz9sTlyjB9MmNPXL5M6aRlqvsscA6HTsbmepuAI4Cjyqb/Bzg/7/tgYn+vtCvQ4j/7Ako9Pl4C\nrkm7PhXquALYCeyn1M53GaU2zyeAF4Efl+1YBvxb2J5NQE/Z8/wtsDX8fL6svAd4LjzmXwkXwSW8\nTR+l1D66EdgQfi6IYLtOBdaH7XoO+KdQflx4028NATk1lL8nzG8Ny48re65rQt23UNZbI819lrFB\nn+ttCvV/NvxsHnndvO+DSf3oylgRkcjluY1eRETqoKAXEYmcgl5EJHIKehGRyCnoRUQip6AXEYmc\ngl5EJHIKehGRyP0/iyyruU+4q4QAAAAASUVORK5CYII=\n",
      "text/plain": [
       "<matplotlib.figure.Figure at 0x7f76dcfdc310>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "#%matplotlib inline\n",
    "#from matplotlib import pyplot as plt\n",
    "#histogr = pd.read_csv('./byposition.txt',sep = '\\t',header = None)\n",
    "#plt.scatter(histogr[6],histogr[7])\n",
    "#plt.scatter()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 2",
   "language": "python",
   "name": "python2"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.12"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
